### DISCUSSION TITLE
how to create block?

### DISCUSSION THREAD

**Discussion by imknight** [area17/twill#1094] [Q&A]
hi all, I have twill based CMS that upgrade from version 1,  it seems like there are some major changes on the blocks, so I m trying to upgrade my old block to the new block. 

Based on the documentation, it seems like there is no longer need to run php artisan twill:blocks and npm run twill-build, but I try to create a new block with the default setting and add it to my block editor, it just blank while for the block content. It is due to my old version 1 setting on Twill config affected it ? or I miss out some steps ?

**Comment by pboivin** [comment]
Hi @imknight,

The new "self-contained" blocks approach from Twill 2.1.0 should be backward compatible with the previous one. I believe your blocks defined in `config/twill.php` shoud still work, but you still need to do a custom build of Twill assets to keep using them.

From what I've seen, blank blocks or repeaters are sometimes caused by outdated assets, so running `php artisan twill:build` could help. If you rewrote all you custom blocks into Blade templates, `php artisan twill:update` will get you back on the built-in assets.

Does this make any sense? Do you think your issue is related to something else?

**Reply by imknight** [reply]
I have fixed it, it seems like you need to manually delete the /resources/views/vendor/twill and publish the new one, it won't get overwritten when you publish the assets

**Reply by pboivin** [reply]
That's great! I'm curious to know more about converting your custom blocks to the new format. Did this go well for you? Did you hit any unexpected behavior in the process?
